paddle/fluid/framework/generator.h

```diff
@@ -59,11 +61,25 @@ struct Generator {
     this->state_.cpu_engine = *engine;
     this->state_.device = -1;
     this->state_.current_seed = seed;
+    this->state_.thread_offset = 0;
     this->engine_ = engine;
     VLOG(4) << "initial seed: " << this->state_.current_seed
             << ", cpu engine: " << &this->state_.cpu_engine;
     this->is_init_py_ = true;  // TODO(zhiqiu): remove it in future
   }
+  explicit Generator(uint64_t seed, uint64_t device_id) {
```

Comment:
`explicit` is not necessary for constructor that has more than one argument?

Reply:
yes